### ISSUE TITLE
Autocomplete, Dropdown, Multiselect : Pressing the escape key once does not close the popup

### ISSUE THREAD

**Issue by nabil-elm** [primefaces/primevue#4212]
### Describe the bug

When the options list is open, pressing the Escape key should close the popup and then move the focus to the autocomplete element. However, when using the NVDA screen reader, the user must press the Escape key twice to close the popup and shift the focus to the autocomplete element. This behavior is due to the focus staying on the options list of p-autocomplete-panel and not returning to <input class="p-autocomplete-input" ...>.
Upon pressing the Escape key once, NVDA's navigation mode switches to browse mode, and the focus remains on the selected option (<li class="p-autocomplete-item p-focus"> of <ul class="p-autocomplete-items">). 
The problem is that the screen reader's browse focus remains at the end of the screen because the HTML code of autocomplete-panel is located after the footer."
If the user does not have the reflex to press the escape key again, it will take several steps to return the focus to the input field of the autocomplete component.
The same behavior for dropdown and multiselect components.

### Reproducer

https://primevue.org/autocomplete/ https://primevue.org/dropdown/ https://primevue.org/multiselect/

### PrimeVue version

3.30.2

### Vue version

3.x

### Language

TypeScript

### Build / Runtime

Vite

### Browser(s)

Chrome, Firefox

### Steps to reproduce the behavior

Upon pressing the Escape key once, NVDA's navigation mode switches to browse mode, and the focus remains on the selected option (<li class="p-autocomplete-item p-focus"> of <ul class="p-autocomplete-items">). 
The problem is that the screen reader's browse focus remains at the end of the screen because the HTML code of autocomplete-panel is located after the footer."
If the user does not have the reflex to press the escape key again, it will take several steps to return the focus to the input field of the autocomplete component.
The same behavior for dropdown and multiselect components.

### Expected behavior

When the options list is open, pressing the Escape key once should close the popup and then move the focus to the autocomplete element.
